static/app/components/demo/demoHeader.tsx

```diff
@@ -16,8 +19,31 @@ export default function DemoHeader() {
   const getStartedUrl = saasOrgSlug
     ? `https://sentry.io/settings/${saasOrgSlug}/billing/checkout/`
     : 'https://sentry.io/signup/';
+
+  const [collapsed, setCollapsed] = useState(Cookies.get('sidebar_collapsed') === '1');
+
+  type Preferences = typeof PreferencesStore.prefs;
+
+  const preferenceUnsubscribe = PreferencesStore.listen(
+    (preferences: Preferences) => onPreferenceChange(preferences),
+    undefined
+  );
+
+  function onPreferenceChange(preferences: Preferences) {
+    if (preferences.collapsed === collapsed) {
```

Comment:
@roggenkemper do we need this? Seems like we could simplify the code and have it work the same if we just always called `setCollapsed`